Sources/PostgresNIO/New/PSQLRowStream.swift

```diff
@@ -31,27 +42,28 @@ final class PSQLRowStream: @unchecked Sendable {
     private let lookupTable: [String: Int]
     private var downstreamState: DownstreamState
     
-    init(rowDescription: [RowDescription.Column],
-         queryContext: ExtendedQueryContext,
-         eventLoop: EventLoop,
-         rowSource: RowSource)
-    {
+    init(
+        source: Source,
+        eventLoop: EventLoop,
+        logger: Logger
+    ) {
         let bufferState: BufferState
-        switch rowSource {
-        case .stream(let dataSource):
+        switch source {
+        case .stream(let rowDescription, let dataSource):
+            self.rowDescription = rowDescription
             bufferState = .streaming(buffer: .init(), dataSource: dataSource)
         case .noRows(.success(let commandTag)):
+            self.rowDescription = []
```

Comment:
In other parts of the codebase (e.g. in `PrepareStatementStateMachine`) we allowed row description to be `nil`. Is there any semantic difference between a `RowDescription?` with a `nil` value and an empty `[RowDescription.Column]`?